manifests/router/connector.pp

```diff
@@ -25,12 +27,25 @@
   Optional[String] $sasl_mech = 'ANONYMOUS',
   Optional[String] $sasl_username = undef,
   Optional[String] $sasl_password = undef,
+  Stdlib::Absolutepath $sasl_password_file = '/etc/qpid-dispatch/connector_pw.conf',
   Optional[Enum['normal', 'inter-router', 'route-container']] $role = undef,
   Optional[String] $ssl_profile = undef,
   Optional[Integer[0]] $idle_timeout = undef,
   String $config_file = $qpid::router::config_file,
 ){
 
+  if defined('$sasl_password') {
```

Comment:
This should be sufficient
```suggestion
  if $sasl_password {
```

Reply:
Thanks. I'm never quite sure about this one.

Reply:
`defined()` checks if the variable is defined. In this case the variable was always defined but with the value `undef`. `defined()` was mostly useful back before there was `$facts` and all facts were on the top scope. That allowed you to have a code path there the fact was not present. The same with global variables, for example set via the ENC. In most cases it's not needed anymore.